app/controllers/cargos_controller.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 class CargosController < ApplicationController
-
+  before_action :authenticate_user!
```

Comment:
Why you moved it from  ApplicationController ?

Reply:
because when I wanted to go to the admin panel at the very beginning I was redirected to user authorization, and only after authorization I was redirected to the administrator page